src/Microsoft.AspNetCore.OData/Formatter/ODataInputFormatter.cs

```diff
@@ -129,7 +129,6 @@ public override Task<InputFormatterResult> ReadRequestBodyAsync(InputFormatterCo
 
                 Action<Exception> logErrorAction = (ex) =>
                 {
-                    throw ex;
```

Comment:
What's the effect of this?

Reply:
This is a logging action only.  It's being used like in ODataInputFormatterHelper.ReadFromStream:

            catch (Exception e)
            {
                logErrorAction(e);
                result = defaultValue;
            }

By throwing, ODataInputFormatterHelper.ReadFromStream was not returning the default value. We should do a better job of logging here.

The AspNet version of this is below where formatterLogger is passed into the function from AspNet and is not null:

                Action<Exception> logErrorAction = (ex) =>
                {
                    if (formatterLogger == null)
                    {
                        throw ex;
                    }

                    formatterLogger.LogError(String.Empty, ex);
                };